server/service/src/catalogue/query_category.rs

```diff
@@ -0,0 +1,35 @@
+use repository::{
+    assets::asset_category::{
+        AssetCategory, AssetCategoryFilter, AssetCategoryRepository, AssetCategorySort,
+    },
+    EqualFilter, PaginationOption, RepositoryError, StorageConnection,
+};
+
+use crate::{get_default_pagination, i64_to_u32, ListError, ListResult};
+
+pub const MAX_LIMIT: u32 = 1000;
+pub const MIN_LIMIT: u32 = 1;
+
+pub fn get_asset_categories(
```

Comment:
It's a bit weird for me that we don't have a service provider and this doesn't take a service context.
I've had a discussion with andrei about it, we think it's probably ok and he thinks it has been done before.
Was there a particular thing you've copied the pattern from, or is this a new innovation?

Reply:
We're all for making development faster and removing roadblocks if they're not helping!

Reply:
Good point!

I saw this in service/sr/temperature_log/mod.rs

The get_temperature_logs takes the connection rather than the context.
I saw that I was only using the connection anyway, so I just passed that value down from the graphql layer instead.

I also import the service layer functions directly rather than using the entire service provider.

This is just an arbitrary choice. I don't really understand the pros and cons of doing it this way.

Shall I refactor to the standard pattern?

Reply:
I think it's ok for now, although maybe it's worth discussing a little bit more. 
The main reasons for the service provider are things like, being able to over-ride functionality for testing (which we very rarely do), often the service layer is where we put in things like extra filters (for example to add a store_id) or to do extra validation logic. If these things aren't needed it does feel like extra overhead. I'll discuss briefly today again before merging I think if we don't follow the pattern we should write up the option and some rules about when to do which pattern maybe?

Reply:
He also mentioned that we want to have tests for our service layer, there's not too much to test here, but having some tests, maybe checking against some of the prepopulated data would be good.
Especially for the `catalogue_item_queries` for `r#type`,  `class`, and `category` filters as they're relying on lookups from different tables so have a bit of extra logic in them...